### PR TITLE
[2.6] Add CPI 1.1.0 (Upstream 1.22.5)

### PR DESCRIPTION
**Pull request by doflamingo721**
Adds:
CPI 1.1.0 (Updates CPI to bring changes from the v1.22.5 upstream)
Port the `permit-os` annotation from https://github.com/rancher/charts/pull/1750

Linked issue:
https://github.com/rancher/rancher/issues/33848
rancher/rke2#2619

### CODE REVIEW

File: release.yaml [hunk 1]
```diff
@@ -44,6 +44,7 @@ rancher-vsphere-csi:
 - 100.1.1+up2.3.0-rancher1
 rancher-vsphere-cpi:
 - 100.1.1+up1.0.1
```
Comment (PennyScissors) [inline release.yaml:46]: Please remove the 100.1.1+up1.0.1 entry and its assets, 100.2.0+up1.1.0 replaces it.

**Review by pennyscissors — CHANGES_REQUESTED** [review]

File: release.yaml (hunk 1 above)
Comment (doflamingo721) [reply release.yaml]: Okay. I made removed the chart entries, can you please review it again?

**Review by pennyscissors — APPROVED** [review]
LGTM, but adding @rosskirkpat to review this because this PR contains the changes he made in PR https://github.com/rancher/charts/pull/1750 which is not merged yet.

**Comment by PennyScissors** [comment]
Also, make sure you update the new charts issue here: https://github.com/rancher/rancher/issues/36185

**Comment by PennyScissors** [comment]
@rosskirkpat removing you because the changes you made were merged already, so no need to review this anymore unless you want to.

@doflamingo721 could you please rebase this PR and resolve the conflicts?

**Comment by aiyengar2** [comment]
Closing in favor of https://github.com/rancher/charts/pull/1787

**Comment by samkulkarni20** [comment]
@snasovich Arvind closed this PR in favor of a similar change, but with an additional commit. Hence reopening this PR won't be a good idea. If you need any specific changes from us, pls let us know & we'll raise a new PR for that.
CC: @doflamingo721 @aiyengar2

**Pull request closed without merging**